packages/core/src/utils/observable.ts

```diff
@@ -0,0 +1,35 @@
+type Listener<T> = (value: T) => void;
+
+interface Observable<T> {
+  subscribe: (listener: Listener<T>) => () => void;
+}
+
+interface ObservableInternal<T> extends Observable<T> {
+  emit: (value: T) => void;
+}
+
+interface ReactiveObject<T> extends Observable<T> {
+  getValue: () => T;
+}
+
+function createObservable<T>(): ObservableInternal<T> {
+  let listeners: Listener<T>[] = [];
+
+  const emit = (value: T): void => {
+    listeners.forEach((listener) => listener(value));
+  };
+
+  const subscribe = (newListener: Listener<T>): () => void => {
+    listeners = [...listeners, newListener];
+
+    return () => { listeners = listeners.filter((listener) => listener !== newListener); };
```

Comment:
We can avoid creating arrays here using `indexOf`+`splice` or maybe using `Set`:

```
let listeners: Se<T>[] = [];
```

```suggestion
    listeners.add(newListener);

    return () => { listeners.delete(newListener) };
```

Reply:
Agree with you.
Changing listeners to the Set instead of the Array will be cool because it solves the problem when the same function is passed to the subscribe multiple times.